### PR TITLE
[VL] Support Spark transform_keys, transform_values function

### PR DESCRIPTION
**Pull request by gaoyangxiaozhu**
## What changes were proposed in this pull request?

(Please fill in changes proposed in this fix)

(Fixes: \#ISSUE-ID)

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

**Comment by gaoyangxiaozhu** [comment]
@rui-mo / @PHILO-HE help review

File: gluten-core/src/main/scala/org/apache/gluten/expression/ExpressionConverter.scala [hunk 1]
```diff
@@ -656,6 +656,22 @@ object ExpressionConverter extends SQLConfHelper with Logging {
...
           c
         )
+      case t: TransformKeys =>
+        // default is `EXCEPTION`
+        val mapKeyDedupPolicy = SQLConf.get.getConf(SQLConf.MAP_KEY_DEDUP_POLICY)
+        if (mapKeyDedupPolicy == SQLConf.MapKeyDedupPolicy.LAST_WIN.toString) {
+          // TODO: Remove after fix ready for
+          //  https://github.com/facebookincubator/velox/issues/10219
+          throw new GlutenNotSupportException(
+            "Not support to handle the cases if duplicate map keys exists after transform" +
```
Comment (PHILO-HE) [inline gluten-core/src/main/scala/org/apache/gluten/expression/ExpressionConverter.scala:666]: Maybe, better to simplify the message like: "LAST_WIN policy is not supported to deduplicate map keys.".

**Review by PHILO-HE — APPROVED** [review]
Looks good! Thanks!

**Comment by gaoyangxiaozhu** [comment]
@PHILO-HE / @rui-mo help merge, thanks

**Pull request merged**